snuba/utils/streams/consumers/backends/abstract.py

```diff
@@ -19,7 +19,7 @@ class ConsumerBackend(ABC, Generic[TStream, TOffset, TValue]):
     def subscribe(
         self,
         topics: Sequence[str],
-        on_assign: Optional[Callable[[Sequence[TStream]], None]] = None,
+        on_assign: Optional[Callable[[Mapping[TStream, TOffset]], None]] = None,
         on_revoke: Optional[Callable[[Sequence[TStream]], None]] = None,
```

Comment:
Do we offsets when revoking a partition as well ?

Reply:
We have them in memory at this point as well, so I suppose we could pass them here for API consistency if you think that makes sense. (I'm not sure what practical benefit they would necessarily have — but I don't think that's necessary a reason for not providing them either.)

Reply:
Discussed this out of band but ended up not making this change — rationale for keeping these types independent is described in this comment: https://github.com/getsentry/snuba/pull/570/commits/1aaf0d6caa27ec319043e42df61ecff6927c4780#diff-969e6da4051dceef9da08c27bb228e9fR42-R49